R/update-regional.R

```diff
@@ -0,0 +1,82 @@
+# Packages -----------------------------------------------------------------
+require(EpiNow2, quietly = TRUE)
+require(covidregionaldata, quietly = TRUE)
+require(data.table, quietly = TRUE)
+require(future, quietly = TRUE)
+require(lubridate, quietly = TRUE)
+
+# Load utils --------------------------------------------------------------
+
+source(here::here("R", "utils.R"))
+
+
+#' update.regional
+#'
+#' @description Processes regional data in an abstract fashion to reduce code duplication
+#' @param region_name String, name of region, used for filepaths
+#' @param region_identifier String, region identifier used by covidregionaldata to get the data #todo audit which countries this differs from name and standardise
+#' @param case_modifier_function Function, passed the cases, should return the cases. Method of modifying the returned data for a specific region if needed
+#' @param generation_time optional overrides for the loaded rds file. If present won't be reloaded from disk.
+#' @param incubation_period optional overrides for the loaded rds file. If present won't be reloaded from disk.
+#' @param reporting_delay optional overrides for the loaded rds file. If present won't be reloaded from disk.
+#' @param cases_region_source string, optional specification of where to get the list of regions from the cases dataset
+#' @param region_scale string, specify the region scale to epinow
+update.regional <- function(region_name, region_identifier, case_modifier_function = NULL, generation_time = NULL, incubation_period = NULL, reporting_delay = NULL, cases_region_source = NULL, region_scale = NULL) {
```

Comment:
I have recently switched from explicitly setting arguments to `NULL` to not giving them a default and using `is.missing()` as the logical check. It makes no functional difference but does give a slightly cleaner look to the function set up (the downside is that it doesn't work well over multiple function levels)